app/uk/gov/hmrc/alcoholdutyreturns/service/ReturnsService.scala

```diff
@@ -70,7 +73,59 @@ class ReturnsService @Inject() (
                                       returnToSubmit,
                                       returnId.appaId
                                     )
+      _                           = cacheRepository.get(returnId).map {
+                                      case Some(ua) =>
+                                        auditReturnSubmitted(
+                                          Some(ua),
+                                          returnToSubmit,
+                                          AdrReturnCreatedDetails.fromReturnCreatedDetails(returnCreatedDetails),
+                                          returnId
+                                        )
+                                      case None     =>
+                                        logger.warn("User answers couldn't be retrieved while auditing return submission")
+                                        auditReturnSubmitted(
+                                          None,
+                                          returnToSubmit,
+                                          AdrReturnCreatedDetails.fromReturnCreatedDetails(returnCreatedDetails),
+                                          returnId
+                                        )
+                                    }
       _                          <- EitherT.right[ErrorResponse](cacheRepository.clearUserAnswersById(returnId))
     } yield returnCreatedDetails
   }
+
+  private def auditReturnSubmitted(
+    userAnswers: Option[UserAnswers],
+    returnToSubmit: ReturnCreate,
+    returnCreatedDetails: AdrReturnCreatedDetails,
+    returnId: ReturnId
+  )(implicit
+    hc: HeaderCarrier
+  ): Unit = {
+    val eventDetail = userAnswers match {
```

Comment:
Can be written shorter:

val eventDetail = AuditReturnSubmitted(
  appaId = returnId.appaId,
  periodKey = returnId.periodKey,
  governmentGatewayId = userAnswers.map(_.internalId),
  governmentGatewayGroupId = userAnswers.map(_.groupId),
  returnSubmittedTime = returnCreatedDetails.processingDate,
  alcoholRegimes = userAnswers.map(_.regimes.regimes),
  requestPayload = returnToSubmit,
  responsePayload = returnCreatedDetails
)